synapse/handlers/auth.py

```diff
@@ -352,7 +354,9 @@ async def validate_user_via_ui_auth(
         requester_user_id = requester.user.to_string()
 
         # Check if we should be ratelimited due to too many previous failed attempts
-        self._failed_uia_attempts_ratelimiter.ratelimit(requester_user_id, update=False)
+        await self._failed_uia_attempts_ratelimiter.ratelimit(
+            requester, requester_user_id, update=False
+        )
```

Comment:
I wonder if we should make the `key` of `ratelimit` optional, and have it default to `requester.user.to_string()`.